roles/openshift_web_console/tasks/smoke_tests.yml

```diff
@@ -0,0 +1,11 @@
+---
+- name: create openshift-web-console-smoke-test project
+  oc_project:
+    name: "{{ openshift_web_console_smoke_test_namespace }}"
+    state: present
+
+- name: create web console smoke test pod 
+  template:
+    namespace: "{{ openshift_web_console_smoke_test_namespace }}"
```

Comment:
The template does not know the `namespace` key. I believe the original intent was to do:
```yaml
- name: Render web console smoke test pod
  template:
    src: "{{ openshift_web_console_smoke_test_pod_template_file }}"
    dest: "{{ openshift_web_console_smoke_test_tmp_location }}/smoke-test-pod.yaml"

- name: Create web console smoke test pod
  oc_obj:
    kind: pod
    name: "{{ openshift_web_console_smoke_test_name }}"
    namespace: "{{ openshift_web_console_smoke_test_namespace }}"
    state: present
    files:
    - "{{ openshift_web_console_smoke_test_tmp_location }}/smoke-test-pod.yaml"
    delete_after: "{{ openshift_web_console_delete_tempfiles }}"
```

Plus those into the `defaults/main.yml`:

```yaml
openshift_web_console_smoke_test_name: origin-web-console-smoke-test-pod
openshift_web_console_delete_tempfiles: True
```

Reply:
After the changes the test pod got deployed:
```
# oc describe pod origin-web-console-smoke-test-pod --namespace openshift-web-console-smoke-test
Name:         origin-web-console-smoke-test-pod
Namespace:    openshift-web-console-smoke-test
Node:         172.16.186.37/172.16.186.37
Start Time:   Wed, 11 Apr 2018 01:55:21 -0400
Labels:       app=origin-web-console-smoke-test
Annotations:  openshift.io/scc=anyuid
              prometheus.io/scheme=http
              prometheus.io/scrape=true
Status:       Running
IP:           10.128.0.139
Containers:
  origin-web-console-smoke-test:
    Container ID:   docker://6081a7dc405c38558a389dcd664b126188924ec8ff2cf4348bca92e0e409712b
    Image:          benjaminapetersen/origin-web-console-smoke-test:latest
    Image ID:       docker-pullable://docker.io/benjaminapetersen/origin-web-console-smoke-test@sha256:434ac7dd9ca1ecb2f06425543f4a494fddb4c0204e649c5d9e5541543008fe39
    Port:           3000/TCP
    State:          Running
      Started:      Wed, 11 Apr 2018 01:56:13 -0400
    Ready:          True
    Restart Count:  0
    Environment:
      CONSOLE_URL:    https://jchaloup-openshift-master-6bk29f-r1:8443/console
      TEST_INTERVAL:  5
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-9rk75 (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          True 
  PodScheduled   True 
Volumes:
  default-token-9rk75:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-9rk75
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  node-role.kubernetes.io/compute=true
Tolerations:     <none>
Events:
  Type    Reason                 Age   From                    Message
  ----    ------                 ----  ----                    -------
  Normal  Scheduled              3m    default-scheduler       Successfully assigned origin-web-console-smoke-test-pod to 172.16.186.37
  Normal  SuccessfulMountVolume  3m    kubelet, 172.16.186.37  MountVolume.SetUp succeeded for volume "default-token-9rk75"
  Normal  Pulling                2m    kubelet, 172.16.186.37  pulling image "benjaminapetersen/origin-web-console-smoke-test:latest"
  Normal  Pulled                 2m    kubelet, 172.16.186.37  Successfully pulled image "benjaminapetersen/origin-web-console-smoke-test:latest"
  Normal  Created                2m    kubelet, 172.16.186.37  Created container
  Normal  Started                2m    kubelet, 172.16.186.37  Started container
```

@jhadvig Anyway to verify the testing pod works properly?

Reply:
```sh
# docker logs 6081a7dc405c
[INFO] Getting certificate for https://jchaloup-openshift-master-6bk29f-r1:8443/console
unable to load certificate
139980689589904:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:696:Expecting: TRUSTED CERTIFICATE
[INFO] Storing obtained certificate
Notice: Trust flag u is set automatically if the private key is present.
certutil: could not decode certificate: SEC_ERROR_INVALID_ARGS: security library: invalid arguments.
[INFO] Launching protractor tests
[INFO] Token eyJhbGciOi***(redacted)
[05:57:14] I/direct - Using ChromeDriver directly...
[05:57:14] I/launcher - Running 1 instances of WebDriver
CONSOLE_URL is https://jchaloup-openshift-master-6bk29f-r1:8443/console
Jasmine started
A Jasmine spec timed out. Resetting the WebDriver Control Flow.

  Openshift login page

    bypass OAuth with service account token
      ✗ should bypass the login flow by injecting a service account token into localStorage
        - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

[05:58:14] E/launcher - this.expectationResultFactory is not a function
[05:58:14] E/launcher - TypeError: this.expectationResultFactory is not a function
    at Suite.onException (/usr/local/lib/node_modules/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:2219:48)
    at QueueRunner.onException (/usr/local/lib/node_modules/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:2330:28)
    at onException (/usr/local/lib/node_modules/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1924:12)
    at Timeout._onTimeout (/usr/local/lib/node_modules/protractor/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1910:11)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
    at Timer.listOnTimeout (timers.js:207:5)
[05:58:14] E/launcher - Process exited with error code 199
[INFO] Server listening to 3000.
[INFO] Exposed enpoints:
[INFO]  /metrics - for scraping test results
[INFO]  /protractor - for triggering origin-web-console smoke test
```

Reply:
Yeah I've check and there is still the old version of the image in the docker hub. We forgot to push the newest.
`jhadvig/origin-web-console-smoke-test:latest` should work